modules/core/src/main/java/org/jsr166/ConcurrentLinkedHashMap.java

```diff
@@ -1536,10 +1536,49 @@ public V replacex(K key, V oldVal, V newVal) {
     }
 
     /**
-     * Removes all of the mappings from this map.
+     * Removes all of the mappings from this map. Performs global locking of the table.
      */
     @Override public void clear() {
-        throw new UnsupportedOperationException();
+
+        try {
+            lockWriteLocks();
+            replaceOldSegments();
+
+        } finally {
+            unlockWriteLocks();
```

Comment:
If the exception will be thrown on lock acquire we will try to unlock locks that never acquired.